### PR TITLE
Create GETDEL and GETRANGE commands

### PR DESCRIPTION
**Pull request by ndelvalle**

### CODE REVIEW

File: src/commands/getrange.rs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        let store = store.lock().unwrap();
+        let value = store.get(&self.key);
+        let bytes = match value {
+            Some(val) => val,
+            None => return Ok(Frame::Bulk(Bytes::new())),
+        };
+
+        let value = String::from_utf8(bytes.to_vec()).unwrap();
+        // TODO: Should we worry about this conversion?
+        let len = value.len() as i64;
```
Comment (gillchristian) [inline src/commands/getrange.rs:36]: If somebody stores a string that is 92233720368547 characters long we have a different problem (if my calculation is not wrong it'd be ~8000 Petabytes).

File: src/commands/getrange.rs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        let start: i64 = start
+            .parse()
+            .map_err(|_| CommandParserError::InvalidCommandArgument { argument: start })?;
+
+        let end: i64 = end
+            .parse()
+            .map_err(|_| CommandParserError::InvalidCommandArgument { argument: end })?;
+
+        Ok(Self { key, start, end })
+    }
```
Comment (gillchristian) [inline src/commands/getrange.rs:78]: Could we use `parser.next_integer` here?

**Review by gillchristian — APPROVED** [review]

File: src/commands/getrange.rs (hunk 1 above)
Comment (ndelvalle) [reply src/commands/getrange.rs:36]: 😁 we should add a frame limit size, so we would never have these kind of issues. That logic I think should live in the codec.

File: src/commands/getrange.rs (hunk 2 above)
Comment (ndelvalle) [reply src/commands/getrange.rs:78]: Of course, I was hoping to use that at some point, and when I got the chance, I didn't hehe.

**Pull request merged**